File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L4/04/Kasus7.py
import numpy as np

def input_matrix(n):
    matrix = np.zeros((n, n))
    for i in range(n):
        for j in range(n):
            matrix[i, j] = float(input())
    return matrix

def sum_major_diagonal(matrix):
    sum_diagonal = np.trace(matrix)
    print(f"Sum of the elements in the major diagonal is {sum_diagonal}")

if __name__ == "__main__":
    n = 4
    print("Enter a 4 by 4 matrix row by row:")
    matrix = input_matrix(n)
    sum_major_diagonal(matrix)

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L4/04/Kasus7.ts
import { readlineSync } from 'readline-sync';

function inputMatrix(matrix: number[][], n: number): void {
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            matrix[i][j] = readlineSync.prompt('Enter a number: ', { float: true });
        }
    }
}

function sumMajorDiagonal(matrix: number[][]): void {
    let sum = 0;
    for (let i = 0; i < matrix.length; i++) {
        sum += matrix[i][i];
    }
    console.log(`Sum of the elements in the major diagonal is ${sum}`);
}

function main(): void {
    const n = 4;
    console.log('Enter a 4 by 4 matrix row by row:');
    const matrix: number[][] = new Array(n).fill(0).map(() => new Array(n).fill(0));
    inputMatrix(matrix, n);
    sumMajorDiagonal(matrix);
}

main();

